Framework/Algorithms/src/FFTSmooth.cpp

```diff
@@ -232,27 +171,17 @@ void FFTSmooth::zero(int n) {
   m_filteredWS =
       API::WorkspaceFactory::Instance().create(m_unfilteredWS, 2, mx, my);
 
-  const Mantid::MantidVec &Yr = m_unfilteredWS->readY(0);
-  const Mantid::MantidVec &Yi = m_unfilteredWS->readY(1);
-  const Mantid::MantidVec &X = m_unfilteredWS->readX(0);
-
-  Mantid::MantidVec &yr = m_filteredWS->dataY(0);
-  Mantid::MantidVec &yi = m_filteredWS->dataY(1);
-  Mantid::MantidVec &xr = m_filteredWS->dataX(0);
-  Mantid::MantidVec &xi = m_filteredWS->dataX(1);
-
-  xr.assign(X.begin(), X.end());
-  xi.assign(X.begin(), X.end());
-  yr.assign(Yr.size(), 0);
-  yi.assign(Yr.size(), 0);
-
-  for (int i = 0; i < ny; i++) {
-    // if (abs(my2-i) < ny2)
-    //{
-    yr[i] = Yr[i];
-    yi[i] = Yi[i];
-    //}
-  }
+  m_filteredWS->setSharedX(0, m_unfilteredWS->sharedX(0));
+  m_filteredWS->setSharedX(1, m_unfilteredWS->sharedX(0));
+
+  m_filteredWS->mutableY(0).assign(m_unfilteredWS->y(0).size(), 0);
+  m_filteredWS->mutableY(1).assign(m_unfilteredWS->y(0).size(), 0);
```

Comment:
I think this is actually not necessary, after `create`, Y and E are initialized to `0` anyway.